commands/sdcli_go_integration

```diff
@@ -27,7 +27,7 @@ INTTESTS=$(go test -v -tags=integration -cover -coverpkg="${PKGS}" -coverprofile
 FAIL="$(echo "${INTTESTS}" | grep 'FAIL')"
```

Comment:
question: @mikerott I am missing some context here as to why exiting right away is desirable, do we want to continue to exit right away? is there a way we can preserve the xunit .xml file for parsing before exiting like the way we do with our unit tests?

Reply:
This code may be the laziest (and maybe unsafe?) way to detect integration test failures, in that it looks at stdout for the word `FAIL` to conclude there must be a test failure.

So this script, as it is written, just runs `go test -tags=integration` and can't tell what went precisely wrong, only that the word `FAIL` showed up.

Since we're already outputting coverage (by `-coverprofile`) we could parse the coverprofile output file instead of looking at stdout for `FAIL`.

Not sure what you mean by "exiting right away".  We need to give back a failure code from the shell script (i.e. `exit 1`) so things like Makefiles will get a failure code propagated out from SDCLI.  Without `exit 1`, the shell prints a message but exits cleanly, docker shuts down cleanly, and the Makefile is non the wiser, also exiting with a success code, and a ci/cd pipeline would no longer be enforcing a quality gate.